Explorer/Assets/Scripts/Global/Dynamic/MainSceneLoader.cs

```diff
@@ -303,6 +305,47 @@ private async UniTask ChangeRealmAsync(CancellationToken ct)
             await realmController.SetRealmAsync(URLDomain.FromString(startingRealm), ct);
         }
 
+        private async UniTask InitializeFeatureFlagsAsync(CancellationToken ct)
+        {
+            try
+            {
+                FeatureFlagOptions options = FeatureFlagOptions.ORG;
+                URLDomain? programArgsUrl = GetUrlFromProgramArgs();
+
+                if (programArgsUrl != null)
+                    options.URL = programArgsUrl.Value;
+
+                // TODO: when the identity is not set, should we request the FFs again after the authentication process?
```

Comment:
I believe it should be implemented somehow as otherwise it may break the flow:
- if it's the first login then what we will receive?
- If the user (during private beta) relogins with the account which is not allowed

at least I suppose `InitializeFeatureFlagsAsync` should be called after `UserInAppInitializationFlow.ExecuteAsync`.
as a resolution for the second issue maybe feature flags should be retrieved as **a part** of `RealUserInitializationFlowController`?

WDYT?

Reply:
In case the user is not authenticated yet we will keep receiving the same data, because we don't have FF assigned to specific users. We are not using that functionality in the remote configuration: https://gh.getunleash.io/reference/activation-strategies#userids
The problem of delaying the fetch of the FF is that any system or service that relies on this data in the startup will not have it available (at least before the authentication) and will become a problem. So i think we should keep doing this call as soon as we can.
On the other part, i added a second fetch after the authentication is resolved, so we can ensure that we get the correct data while in-world: https://github.com/decentraland/unity-explorer/pull/1279/commits/3ff7478e17f704ee5e2cbed25bdc608508c55726
The request might be big, but it should not add a performance degradation

Reply:
You are right, makes sense, thank you